tools/tokengen/src/server/server_impl.rs

```diff
@@ -0,0 +1,308 @@
+use axum::{
+    extract::{Extension, Json},
+    routing::post,
+    Router,
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    env,
+    error::Error,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use uuid::Uuid;
+
+use crate::{generate_token, ApplicationProtocol, RecordingOperation, SubCommandArgs};
+
+pub(crate) fn create_router(provisioner_key_path: Arc<PathBuf>) -> Router {
+    Router::new()
+        .route("/forward", post(forward_handler))
+        .route("/rendezvous", post(rendezvous_handler))
+        .route("/rdp_tls", post(rdp_tls_handler))
+        .route("/scope", post(scope_handler))
+        .route("/jmux", post(jmux_handler))
+        .route("/jrec", post(jrec_handler))
+        .route("/kdc", post(kdc_handler))
+        .route("/jrl", post(jrl_handler))
+        .route("/netscan", post(netscan_handler))
+        .layer(Extension(provisioner_key_path))
+}
+
+pub(crate) async fn get_provisioner_key_path() -> Result<Arc<PathBuf>, Box<dyn Error>> {
+    let config_dir = env::var("DGATEWAY_CONFIG_PATH").expect("DGATEWAY_CONFIG_PATH environment variable not set");
+
+    let gateway_json_path = Path::new(&config_dir).join("gateway.json");
+
+    let gateway_json_contents = tokio::fs::read_to_string(&gateway_json_path).await?;
+    let gateway_config: serde_json::Value = serde_json::from_str(&gateway_json_contents)?;
+
+    let provisioner_private_key_file = gateway_config
+        .get("ProvisionerPrivateKeyFile")
+        .ok_or("ProvisionerPrivateKeyFile not found in gateway.json")?
+        .as_str()
+        .ok_or("ProvisionerPrivateKeyFile is not a string")?;
+
+    let provisioner_key_path = PathBuf::from(provisioner_private_key_file);
+    let provisioner_key_path = if provisioner_key_path.is_relative() {
+        gateway_json_path.parent().unwrap().join(provisioner_key_path)
+    } else {
+        provisioner_key_path
+    };
+
+    Ok(Arc::new(provisioner_key_path))
+}
+
+#[derive(Deserialize)]
+pub(crate) struct CommonRequest {
+    #[serde(default)]
+    validity_duration: Option<String>,
+    #[serde(default)]
+    kid: Option<String>,
+    #[serde(default)]
+    delegation_key_path: Option<String>,
```

Comment:
I think this should not be part of the request. Why not proceed like for `provisioner_key_path`?